### DISCUSSION TITLE
Is there a known issue with read storage and web3?

### DISCUSSION THREAD

**Discussion by galactictourist** [crytic/slither#1747] [General]
submitted a bug already but am desperate to make this work - no matter what I do I cannot get slither to recognize my web3 install.

(slither-dev) iMac:slither sovryn73$ slither-read-storage bsc:0xa1faa113cbE53436Df28FF0aEe54275c13B40975 --json storage_layout.json --rpc-url https://bsc-mainnet-rpc.allthatnode.com 
ERROR: in order to use slither-read-storage, you need to install web3
$ pip3 install web3 --user

this over and over

**Comment by 0xalpharush** [comment]
The try-catch is catching more errors than it should as pointed out here (https://github.com/crytic/slither/pull/1710/). I believe this is causes by web3.py's [lack of support](https://github.com/ethereum/web3.py/issues/2704) for python3.11, but lmk if you're not using 3.11. There's pull request to fix the issue here https://github.com/crytic/slither/pull/1743. In the meantime, I think the only solution is to downgrade your python version

**Reply by galactictourist** [reply]
Yep, using 3.11.2 - so just downgrade and rerun the command and all good? What version is best to downgrade to? Thank you!

**Reply by 0xalpharush** [reply]
I believe 3.10 should suffice

**Reply by galactictourist** [reply]
thanks

**Comment by maxcoto** [comment]
I have this same problem with python 3.9 and 3.10 too.
slither version 0.9.2
```
slither-read-storage 0x8ad599c3a0ff1de082011efddc58f1908eb6e6d8
ERROR: in order to use slither-read-storage, you need to install web3
$ pip3 install web3 --user
```
